packages/app/src/api/visualization.js

```diff
@@ -1,46 +1,36 @@
-import { getInstance } from 'd2'
 import { getFieldsStringByType } from '../modules/fields'
 
-export const apiFetchVisualization = (type, id) =>
-    getInstance().then(d2 =>
-        d2.models[type].get(id, {
-            fields: getFieldsStringByType(type),
-        })
-    )
+export const apiFetchVisualization = (dataEngine, id) => {
+    const visualizationQuery = {
```

Comment:
To be pedantic, this query should be "static" (constant outside the function) and accept an `id` dynamic variable:

```js
const visualizationQuery = {
  visualization: {
    resource: 'visualizations',
    id: ({ id }) => id,
    params: {
      fields: getFieldsStringByType('visualization'),
    },
  },
}

export const apiFetchVisualization = (dataEngine, id) => {
  return dataEngine.query(visualizationQuery, { variables: { id } })
}
```
This is a pretty minor point, though, since it's more relevant when used in a declarative context (React hooks)